vm/compiler/benches/coinbase_puzzle.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+#![allow(clippy::single_element_loop)]
+
+#[macro_use]
+extern crate criterion;
+
+use console::{account::*, network::Testnet3};
+use snarkvm_compiler::{CoinbasePuzzle, EpochChallenge, PuzzleConfig};
+
+use criterion::Criterion;
+use rand::{self, thread_rng, CryptoRng, RngCore};
```

Comment:
```suggestion
use rand::{self, CryptoRng, RngCore};
```
we should instead be using `snarkvm_utilities::TestRng` for greater test/benchmark performance and easier reproducibility; this applies to all other tests and benchmarks